src/opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp

```diff
@@ -0,0 +1,916 @@
+  ParseContext::PARSE_WGNAME_SPACE setting is set to InputError::IGNORE that
+  means that we do not inform the user about "our fix", but it is *not* possible
+  to configure the parser to leave the spaces intact.
+*/
+std::string trim_wgname(const DeckKeyword& keyword,
+                        const std::string& wgname_arg,
+                        const ParseContext& parseContext,
+                        ErrorGuard& errors)
+{
+    std::string wgname = trim_copy(wgname_arg);
+    if (wgname != wgname_arg)  {
+        const auto& location = keyword.location();
+        std::string msg_fmt = fmt::format("Problem with keyword {{keyword}}\n"
+                                          "In {{file}} line {{line}}\n"
+                                          "Illegal space in {} when defining WELL/GROUP.", wgname_arg);
+        parseContext.handleError(ParseContext::PARSE_WGNAME_SPACE, msg_fmt, location, errors);
+    }
+    return wgname;
+}
+
+void handleWCONHIST(HandlerContext& handlerContext)
+{
+    for (const auto& record : handlerContext.keyword) {
+        const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
+        const auto well_names = handlerContext.wellNames(wellNamePattern, false);
+
+        const Well::Status status = WellStatusFromString(record.getItem("STATUS").getTrimmedString(0));
+
+        for (const auto& well_name : well_names) {
+            handlerContext.updateWellStatus(well_name, status,
+                                            handlerContext.keyword.location());
+
+            std::optional<VFPProdTable::ALQ_TYPE> alq_type;
+            auto well2 = handlerContext.state().wells.get( well_name );
+            const bool switching_from_injector = !well2.isProducer();
+            auto properties = std::make_shared<Well::WellProductionProperties>(well2.getProductionProperties());
+            bool update_well = false;
+
+            auto table_nr = record.getItem("VFP_TABLE").get< int >(0);
+            if (record.getItem("VFP_TABLE").defaultApplied(0)) {
+                table_nr = properties->VFPTableNumber;
+            }
+
+            if (table_nr != 0) {
+                const auto& vfpprod = handlerContext.state().vfpprod;
+                if (vfpprod.has(table_nr)) {
+                    alq_type = handlerContext.state().vfpprod(table_nr).getALQType();
+                } else {
+                    std::string reason = fmt::format("Problem with well:{} VFP table: {} not defined", well_name, table_nr);
+                    throw OpmInputError(reason, handlerContext.keyword.location());
+                }
+            }
+            double default_bhp;
+            if (handlerContext.state().bhp_defaults.get().prod_target) {
+                default_bhp = *handlerContext.state().bhp_defaults.get().prod_target;
+            } else {
+                default_bhp = UnitSystem::newMETRIC().to_si(UnitSystem::measure::pressure,
```

Comment:
You should `#include <opm/input/eclipse/Units/UnitSystem.hpp>` for this statement.

Reply:
indeed